frame/contracts/src/exec.rs

```diff
@@ -306,6 +310,22 @@ pub trait Ext: sealing::Sealed {
 
 	/// Returns a nonce that is incremented for every instantiated contract.
 	fn nonce(&mut self) -> u64;
+
+	/// Add a contract dependency, This is useful for contract delegation, to make sure that the
+	/// delegated contract is not removed while it is still in use.
+	/// This will increase the reference count of the code hash, and charge a deposit of 30% of the
+	/// code deposit.
+	///
+	/// Returns an error if we have reached the maximum number of dependencies, or the dependency
+	/// already exists.
+	fn add_dependency(&mut self, code_hash: CodeHash<Self::T>) -> Result<(), DispatchError>;
```

Comment:
I am thinking we should  these functions `add_delegate_dependency` & `remove_delegate_dependency` just to make it clearer that this is mainly useful to lock a delegate_dependency. What do you think?